diff --git a/lib/c/c_Client.cc b/lib/c/c_Client.cc
index 9f27538..af4aa76 100644
--- a/lib/c/c_Client.cc
+++ b/lib/c/c_Client.cc
@@ -21,6 +21,8 @@

 #include "c_structs.h"

+#include <unistd.h>
+
 pulsar_client_t *pulsar_client_create(const char *serviceUrl,
                                       const pulsar_client_configuration_t *clientConfiguration) {
     pulsar_client_t *c_client = new pulsar_client_t;
@@ -63,6 +65,9 @@ void pulsar_client_create_producer_async(pulsar_client_t *client, const char *to
                                                   std::placeholders::_2, callback, ctx));
 }

+// The sleep()s prevent a race-condition between function return and the logger
+// FFI callback. They are a dirty work-around and should be removed once we
+// solved the root-cause.
 pulsar_result pulsar_client_subscribe(pulsar_client_t *client, const char *topic,
                                       const char *subscriptionName,
                                       const pulsar_consumer_configuration_t *conf,
@@ -73,8 +78,10 @@ pulsar_result pulsar_client_subscribe(pulsar_client_t *client, const char *topic
     if (res == pulsar::ResultOk) {
         (*c_consumer) = new pulsar_consumer_t;
         (*c_consumer)->consumer = consumer;
+        sleep(1);
         return pulsar_result_Ok;
     } else {
+        sleep(1);
         return (pulsar_result)res;
     }
 }
